### PR TITLE
Dev rep server

### PR DESCRIPTION
**Pull request by micahu**
Adds configurations for development work on the reputation server

### CODE REVIEW

File: migrations/2_deploy_contracts.js [hunk 1]
```diff
@@ -52,11 +57,29 @@ module.exports = function (deployer, network, accounts) {
...
+    .then(() => network === 'devRepServer' ? a.addNetworkMember(actors.mfg, false, true, false, 'auditMan', { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? t.transfer(actors.mfg, 300000000000000000000000000, { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...registrate, activate
+    .then(() => network === 'devRepServer' ? t.approve(a.address, 300000000000000000000000000, { from: actors.mfg }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Mfg approved tokens for Atonomi to withdraw:', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.registerAndActivateDevice(deviceId, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Device registered and activated by mfg:'/*, deviceId*/, tx.receipt.status) : null)
+    // ...reputate
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: Remove this.  This reputation writes will be tested through the API

File: migrations/2_deploy_contracts.js [hunk 2]
```diff
@@ -52,11 +57,29 @@ module.exports = function (deployer, network, accounts) {
...
+    .then(() => network === 'devRepServer' ? a.addNetworkMember(actors.irnNode, false, false, true, '', { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Network member added (owner):', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.addNetworkMember(actors.mfg, false, true, false, 'auditMan', { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? t.transfer(actors.mfg, 300000000000000000000000000, { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...registrate, activate
+    .then(() => network === 'devRepServer' ? t.approve(a.address, 300000000000000000000000000, { from: actors.mfg }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Mfg approved tokens for Atonomi to withdraw:', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.registerAndActivateDevice(deviceId, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: You will still need two devices for your testing.  One that is just registered and one that is activated

File: migrations/2_deploy_contracts.js [hunk 3]
```diff
@@ -52,11 +57,29 @@ module.exports = function (deployer, network, accounts) {
...
-    .then(() => network === 'web' ? t.releaseTokenTransfer({from: owner}) : null)
-    .then((tx) => network === 'web' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+    .then((tx) => network === 'web' || network === 'devRepServer' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'devRepServer' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'devRepServer' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'devRepServer' ? a.addNetworkMember(actors.irnNode, false, false, true, '', { from: owner }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Network member added (owner):', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.addNetworkMember(actors.mfg, false, true, false, 'auditMan', { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: Keep MFGID to 4 characters to be consistent with real data.

File: migrations/2_deploy_contracts.js [hunk 4]
```diff
@@ -52,11 +57,29 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'devRepServer' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? t.transfer(actors.mfg, 300000000000000000000000000, { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: use multiplier for readability.

File: migrations/2_deploy_contracts.js [hunk 5]
```diff
@@ -52,11 +57,29 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'devRepServer' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...registrate, activate
+    .then(() => network === 'devRepServer' ? t.approve(a.address, 300000000000000000000000000, { from: actors.mfg }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: use multiplier for readability

File: package.json [hunk 6]
```diff
@@ -47,6 +47,7 @@
     "ganache-cli": "ganache-cli -p 8546 -e 1000 -a 15 -l 990000000000000",
     "deploy": "truffle migrate --reset",
     "deploy:web": "truffle migrate --reset --network web && cp -rf ./build/. ../admin-portal/build/",
+    "deploy:devRepServer": "truffle migrate --reset --network devRepServer && cp -rf ./build/. ../admin-portal/build/",
```
Comment (filcap) [inline package.json]: `deploy:rep`, keep the scripts and code shorter

**Comment by micahu** [comment]
thanks. took your advice, see latest commit.

File: migrations/2_deploy_contracts.js [hunk 7]
```diff
@@ -15,6 +16,10 @@ module.exports = function (deployer, network, accounts) {
   const repReward = 1 * multiplier
   const reputationShare = 20
   const blockThreshold = 5760 // assuming 15s blocks, 1 write per day
+  const deviceId = 'apple-iphone1'
+  const deviceIdHash = web3Utils.soliditySha3({t: 'bytes32', v: web3.fromAscii(deviceId)})
+  const devicePublicKey = '0x9c274091da1ce47bd321f272d66b6e5514fb82346d7992e2d1a3eefdeffed791'
+  const mfgId = 'DevMfg'
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: Keep to 4 characters to match what CENTRI expects

File: migrations/2_deploy_contracts.js [hunk 8]
```diff
@@ -52,11 +57,31 @@ module.exports = function (deployer, network, accounts) {
...
     })
-    .then((tx) => network === 'web' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
-    .then(() => network === 'web' ? t.releaseTokenTransfer({from: owner}) : null)
-    .then((tx) => network === 'web' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'reputation' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.irnNode, false, false, true, 'NODE', { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: irn nodes, leave mfgid as empty string

File: migrations/2_deploy_contracts.js [hunk 9]
```diff
@@ -42,7 +47,7 @@ module.exports = function (deployer, network, accounts) {
           true, // irn admin
           true, // manufacturer
           true, // irn node
-          'DEV', // manufacturer id
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: Discard this change, so not to break web deployment

File: migrations/2_deploy_contracts.js [hunk 10]
```diff
@@ -52,11 +57,31 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'reputation' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.irnNode, false, false, true, 'NODE', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (owner):', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.mfg, false, true, false, 'APPL', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? t.transfer(actors.mfg, 300000000 * multiplier, { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: that is a lot of tokens.   just test with 100

File: migrations/2_deploy_contracts.js [hunk 11]
```diff
@@ -52,11 +57,31 @@ module.exports = function (deployer, network, accounts) {
...
-    .then(() => network === 'web' ? t.releaseTokenTransfer({from: owner}) : null)
-    .then((tx) => network === 'web' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'reputation' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.irnNode, false, false, true, 'NODE', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (owner):', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.mfg, false, true, false, 'APPL', { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: use your mfgId from above

File: migrations/2_deploy_contracts.js [hunk 12]
```diff
@@ -52,11 +57,31 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'reputation' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? t.transfer(actors.mfg, 300000000 * multiplier, { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...registrate, activate
+    .then(() => network === 'reputation' ? t.approve(a.address, 300000000 * multiplier, { from: actors.mfg }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: just approve the amount that is needed for two registrations and one activation

File: migrations/2_deploy_contracts.js [hunk 13]
```diff
@@ -52,11 +57,31 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'reputation' ? console.log('Mfg approved tokens for Atonomi to withdraw:', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.registerDevice(deviceIdHash, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Device registered by mfg:', deviceIdHash, tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? a.registerAndActivateDevice(deviceId, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Device registered and activated by mfg:', tx.receipt.status) : null)
+    // ...reputate
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: Remove all the reputation score updates here.

File: migrations/2_deploy_contracts.js [hunk 14]
```diff
@@ -52,11 +56,26 @@ module.exports = function (deployer, network, accounts) {
...
-    .then((tx) => network === 'web' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
-    .then(() => network === 'web' ? t.releaseTokenTransfer({from: owner}) : null)
-    .then((tx) => network === 'web' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'reputation' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.irnNode, false, false, true, '', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (owner):', tx.receipt.status) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: typo (owner) -> (irnnode)

File: migrations/2_deploy_contracts.js [hunk 15]
```diff
@@ -52,11 +56,26 @@ module.exports = function (deployer, network, accounts) {
...
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.mfg, false, true, false, 'APPL', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? t.transfer(actors.mfg, 3 * multiplier, { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...register, activate
+    .then(() => network === 'reputation' ? t.approve(a.address, 3 * multiplier, { from: actors.mfg }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Mfg approved tokens for Atonomi to withdraw:', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.registerDevice(deviceIdHash, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
+    .then((tx) => network === 'devRepServer' ? console.log('Device registered by mfg:', deviceIdHash, tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? a.registerAndActivateDevice(deviceId, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: You will want to use a different device id for this one.

File: migrations/2_deploy_contracts.js [hunk 16]
```diff
@@ -52,11 +56,26 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Owner set as release agent:', tx.receipt.status) : null)
+    .then(() => network === 'web' || network === 'reputation' ? t.releaseTokenTransfer({from: owner}) : null)
+    .then((tx) => network === 'web' || network === 'reputation' ? console.log('Token transfers enabled:', tx.receipt.status) : null)
+
+    // sets up development environment testing for rep server
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.irnNode, false, false, true, '', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (owner):', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? a.addNetworkMember(actors.mfg, false, true, false, 'APPL', { from: owner }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Network member added (mfg): ', tx.receipt.status) : null)
+    .then(() => network === 'reputation' ? t.transfer(actors.mfg, 3 * multiplier, { from: owner }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: use `regFee` and actFee`, hardcoded could break if we change those later

File: migrations/2_deploy_contracts.js [hunk 17]
```diff
@@ -52,11 +56,26 @@ module.exports = function (deployer, network, accounts) {
...
+    .then((tx) => network === 'reputation' ? console.log('Tokens transferred from owner to mfg:', tx.receipt.status) : null)
+    // ...register, activate
+    .then(() => network === 'reputation' ? t.approve(a.address, 3 * multiplier, { from: actors.mfg }) : null)
+    .then((tx) => network === 'reputation' ? console.log('Mfg approved tokens for Atonomi to withdraw:', tx.receipt.status) : null)
+    .then(() => network === 'devRepServer' ? a.registerDevice(deviceIdHash, 'smartphone', devicePublicKey, { from: actors.mfg }) : null)
```
Comment (filcap) [inline migrations/2_deploy_contracts.js]: update if condition

**Comment by micahu** [comment]
i also added a logging item to show allowance during the deploy script. this is strictly informational but i found it helpful to understand the state on deployment without having to open up a geth console.

**Pull request merged**